### PR TITLE
Feat/iopr 4229

### PR DESCRIPTION
**Pull request by gbeaumont93**

### CODE REVIEW

File: internal/data_sources/physical_port_data_source.go [hunk 1]
```diff
@@ -0,0 +1,319 @@
...
+		if matches {
+			result = append(result, physicalPort)
+		}
+	}
+	return result
+}
+
+func matchesFilter(physicalPortValue string, operator string, filterFieldValue []types.String) bool {
+	switch operator {
+	case "=":
```
Comment (Barbeek) [inline internal/data_sources/physical_port_data_source.go:308]: You can use the filter type const (common.go)

File: internal/data_sources/physical_port_data_source.go [hunk 2]
```diff
@@ -0,0 +1,319 @@
...
+				}
+			case "priceMrc":
+				if !matchesFilter(strconv.Itoa(physicalPort.Product.PriceMRC), filter.Operator.String(), filterValue) {
+					matches = false
+				}
+			case "priceNrc":
+				if !matchesFilter(strconv.Itoa(physicalPort.Product.PriceNRC), filter.Operator.String(), filterValue) {
+					matches = false
+				}
+			case "costMrc":
```
Comment (Barbeek) [inline internal/data_sources/physical_port_data_source.go:289]: Is the cost not what Intercloud pays ? I'm not sure users would want to filter on that. According to me the filter should only be done on name, physical port id,  location and priceMrc

File: internal/data_sources/physical_port_data_source.go [hunk 3]
```diff
@@ -0,0 +1,319 @@
...
+		}
+		if matches {
+			result = append(result, physicalPort)
+		}
+	}
+	return result
+}
+
+func matchesFilter(physicalPortValue string, operator string, filterFieldValue []types.String) bool {
+	switch operator {
```
Comment (Barbeek) [inline internal/data_sources/physical_port_data_source.go:307]: Is there a reason why OR is not checked ?

File: internal/data_sources/physical_port_data_source.go (hunk 2 above)
Comment (gbeaumont93) [reply internal/data_sources/physical_port_data_source.go]: I remove them from the schema but maybe we should remove them from other datasource and resources

File: internal/data_sources/filters/autonomi.go [hunk 4]
```diff
@@ -0,0 +1,105 @@
...
+			if !filter.match(&physicalPort) {
+				continue mainloop
+			}
+		}
+		result = append(result, physicalPort)
+	}
+	return result, nil
+}
+
+func (fg *filterAutonomi) match(physicalPort *autonomisdkmodel.PhysicalPort) bool {
```
[the remark this answers is not in the document]
Comment (gbeaumont93) [reply internal/data_sources/filters/autonomi.go:76]: un filtre sur l'ID ?

[the remark this answers is not in the document]
Comment (gbeaumont93) [reply internal/data_sources/filters/autonomi.go:76]: done

**Review by Barbeek — APPROVED** [review]

**Pull request merged**